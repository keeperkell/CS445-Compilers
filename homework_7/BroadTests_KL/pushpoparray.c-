int a, aa[100];

int dog(int pa[])
 begin 
    pa[66];
    a <- pa[22];
    pa[88]++;
    output(a);
    output(pa[88]);
    return pa[22]+pa[44];
 end 


main()
 begin 
   int b, bb[100];

   aa[11] <- 11;
   bb[22] <- 22;
   aa[33] <- 33;
   bb[44] <- 44;
   aa[55] <- 55;
   bb[66] <- 66;
   aa[77] <- 77;
   bb[88] <- 88;
   aa[99] <- 99;
   
   output(aa[11]);
   output(bb[22]);
   output(aa[99]++);
   output(aa[99]);
   outnl();

   output(a <- aa[33]);
   output(a);
   outnl();

   output(bb[66-44]);
   output(aa[77] * bb[88]);
   outnl();

   output(dog(bb));
   output(bb[88]);
   output(bb[dog(bb)]);
   outnl();
 end 



