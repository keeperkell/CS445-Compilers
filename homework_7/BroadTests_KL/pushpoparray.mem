WARNING(16): The variable 'b' seems not to be used.
Func: input returns type int [mem: Global loc: 0 size: -2] [line: -1]
Sibling: 1  Func: output returns type void [mem: Global loc: 0 size: -3] [line: -1]
.   Child: 0  Parm: *dummy* of type int [mem: Parameter loc: -2 size: 1] [line: -1]
Sibling: 2  Func: inputb returns type bool [mem: Global loc: 0 size: -2] [line: -1]
Sibling: 3  Func: outputb returns type void [mem: Global loc: 0 size: -3] [line: -1]
.   Child: 0  Parm: *dummy* of type bool [mem: Parameter loc: -2 size: 1] [line: -1]
Sibling: 4  Func: inputc returns type char [mem: Global loc: 0 size: -2] [line: -1]
Sibling: 5  Func: outputc returns type void [mem: Global loc: 0 size: -3] [line: -1]
.   Child: 0  Parm: *dummy* of type char [mem: Parameter loc: -2 size: 1] [line: -1]
Sibling: 6  Func: outnl returns type void [mem: Global loc: 0 size: -2] [line: -1]
Sibling: 7  Var: a of type int [mem: Global loc: 0 size: 1] [line: 1]
Sibling: 8  Var: aa of array of type int [mem: Global loc: -2 size: 101] [line: 1]
Sibling: 9  Func: dog returns type int [mem: Global loc: 0 size: -3] [line: 3]
.   Child: 0  Parm: pa of array of type int [mem: Parameter loc: -2 size: 1] [line: 3]
.   Child: 1  Compound [mem: None loc: 0 size: -3] [line: 4]
.   .   Child: 1  Op: [ of type int [line: 5]
.   .   .   Child: 0  Id: pa of array of type int [mem: Parameter loc: -2 size: 1] [line: 5]
.   .   .   Child: 1  Const 66 of type int [line: 5]
.   .   Sibling: 1  Assign: <- of type int [line: 6]
.   .   .   Child: 0  Id: a of type int [mem: Global loc: 0 size: 1] [line: 6]
.   .   .   Child: 1  Op: [ of type int [line: 6]
.   .   .   .   Child: 0  Id: pa of array of type int [mem: Parameter loc: -2 size: 1] [line: 6]
.   .   .   .   Child: 1  Const 22 of type int [line: 6]
.   .   Sibling: 2  Assign: ++ of type int [line: 7]
.   .   .   Child: 0  Op: [ of type int [line: 7]
.   .   .   .   Child: 0  Id: pa of array of type int [mem: Parameter loc: -2 size: 1] [line: 7]
.   .   .   .   Child: 1  Const 88 of type int [line: 7]
.   .   Sibling: 3  Call: output of type void [line: 8]
.   .   .   Child: 0  Id: a of type int [mem: Global loc: 0 size: 1] [line: 8]
.   .   Sibling: 4  Call: output of type void [line: 9]
.   .   .   Child: 0  Op: [ of type int [line: 9]
.   .   .   .   Child: 0  Id: pa of array of type int [mem: Parameter loc: -2 size: 1] [line: 9]
.   .   .   .   Child: 1  Const 88 of type int [line: 9]
.   .   Sibling: 5  Return [line: 10]
.   .   .   Child: 0  Op: + of type int [line: 10]
.   .   .   .   Child: 0  Op: [ of type int [line: 10]
.   .   .   .   .   Child: 0  Id: pa of array of type int [mem: Parameter loc: -2 size: 1] [line: 10]
.   .   .   .   .   Child: 1  Const 22 of type int [line: 10]
.   .   .   .   Child: 1  Op: [ of type int [line: 10]
.   .   .   .   .   Child: 0  Id: pa of array of type int [mem: Parameter loc: -2 size: 1] [line: 10]
.   .   .   .   .   Child: 1  Const 44 of type int [line: 10]
Sibling: 10  Func: main returns type void [mem: Global loc: 0 size: -2] [line: 14]
.   Child: 1  Compound [mem: None loc: 0 size: -104] [line: 15]
.   .   Child: 0  Var: b of type int [mem: Local loc: -2 size: 1] [line: 16]
.   .   Sibling: 1  Var: bb of array of type int [mem: Local loc: -4 size: 101] [line: 16]
.   .   Child: 1  Assign: <- of type int [line: 18]
.   .   .   Child: 0  Op: [ of type int [line: 18]
.   .   .   .   Child: 0  Id: aa of array of type int [mem: Global loc: -2 size: 101] [line: 18]
.   .   .   .   Child: 1  Const 11 of type int [line: 18]
.   .   .   Child: 1  Const 11 of type int [line: 18]
.   .   Sibling: 1  Assign: <- of type int [line: 19]
.   .   .   Child: 0  Op: [ of type int [line: 19]
.   .   .   .   Child: 0  Id: bb of array of type int [mem: Local loc: -4 size: 101] [line: 19]
.   .   .   .   Child: 1  Const 22 of type int [line: 19]
.   .   .   Child: 1  Const 22 of type int [line: 19]
.   .   Sibling: 2  Assign: <- of type int [line: 20]
.   .   .   Child: 0  Op: [ of type int [line: 20]
.   .   .   .   Child: 0  Id: aa of array of type int [mem: Global loc: -2 size: 101] [line: 20]
.   .   .   .   Child: 1  Const 33 of type int [line: 20]
.   .   .   Child: 1  Const 33 of type int [line: 20]
.   .   Sibling: 3  Assign: <- of type int [line: 21]
.   .   .   Child: 0  Op: [ of type int [line: 21]
.   .   .   .   Child: 0  Id: bb of array of type int [mem: Local loc: -4 size: 101] [line: 21]
.   .   .   .   Child: 1  Const 44 of type int [line: 21]
.   .   .   Child: 1  Const 44 of type int [line: 21]
.   .   Sibling: 4  Assign: <- of type int [line: 22]
.   .   .   Child: 0  Op: [ of type int [line: 22]
.   .   .   .   Child: 0  Id: aa of array of type int [mem: Global loc: -2 size: 101] [line: 22]
.   .   .   .   Child: 1  Const 55 of type int [line: 22]
.   .   .   Child: 1  Const 55 of type int [line: 22]
.   .   Sibling: 5  Assign: <- of type int [line: 23]
.   .   .   Child: 0  Op: [ of type int [line: 23]
.   .   .   .   Child: 0  Id: bb of array of type int [mem: Local loc: -4 size: 101] [line: 23]
.   .   .   .   Child: 1  Const 66 of type int [line: 23]
.   .   .   Child: 1  Const 66 of type int [line: 23]
.   .   Sibling: 6  Assign: <- of type int [line: 24]
.   .   .   Child: 0  Op: [ of type int [line: 24]
.   .   .   .   Child: 0  Id: aa of array of type int [mem: Global loc: -2 size: 101] [line: 24]
.   .   .   .   Child: 1  Const 77 of type int [line: 24]
.   .   .   Child: 1  Const 77 of type int [line: 24]
.   .   Sibling: 7  Assign: <- of type int [line: 25]
.   .   .   Child: 0  Op: [ of type int [line: 25]
.   .   .   .   Child: 0  Id: bb of array of type int [mem: Local loc: -4 size: 101] [line: 25]
.   .   .   .   Child: 1  Const 88 of type int [line: 25]
.   .   .   Child: 1  Const 88 of type int [line: 25]
.   .   Sibling: 8  Assign: <- of type int [line: 26]
.   .   .   Child: 0  Op: [ of type int [line: 26]
.   .   .   .   Child: 0  Id: aa of array of type int [mem: Global loc: -2 size: 101] [line: 26]
.   .   .   .   Child: 1  Const 99 of type int [line: 26]
.   .   .   Child: 1  Const 99 of type int [line: 26]
.   .   Sibling: 9  Call: output of type void [line: 28]
.   .   .   Child: 0  Op: [ of type int [line: 28]
.   .   .   .   Child: 0  Id: aa of array of type int [mem: Global loc: -2 size: 101] [line: 28]
.   .   .   .   Child: 1  Const 11 of type int [line: 28]
.   .   Sibling: 10  Call: output of type void [line: 29]
.   .   .   Child: 0  Op: [ of type int [line: 29]
.   .   .   .   Child: 0  Id: bb of array of type int [mem: Local loc: -4 size: 101] [line: 29]
.   .   .   .   Child: 1  Const 22 of type int [line: 29]
.   .   Sibling: 11  Call: output of type void [line: 30]
.   .   .   Child: 0  Assign: ++ of type int [line: 30]
.   .   .   .   Child: 0  Op: [ of type int [line: 30]
.   .   .   .   .   Child: 0  Id: aa of array of type int [mem: Global loc: -2 size: 101] [line: 30]
.   .   .   .   .   Child: 1  Const 99 of type int [line: 30]
.   .   Sibling: 12  Call: output of type void [line: 31]
.   .   .   Child: 0  Op: [ of type int [line: 31]
.   .   .   .   Child: 0  Id: aa of array of type int [mem: Global loc: -2 size: 101] [line: 31]
.   .   .   .   Child: 1  Const 99 of type int [line: 31]
.   .   Sibling: 13  Call: outnl of type void [line: 32]
.   .   Sibling: 14  Call: output of type void [line: 34]
.   .   .   Child: 0  Assign: <- of type int [line: 34]
.   .   .   .   Child: 0  Id: a of type int [mem: Global loc: 0 size: 1] [line: 34]
.   .   .   .   Child: 1  Op: [ of type int [line: 34]
.   .   .   .   .   Child: 0  Id: aa of array of type int [mem: Global loc: -2 size: 101] [line: 34]
.   .   .   .   .   Child: 1  Const 33 of type int [line: 34]
.   .   Sibling: 15  Call: output of type void [line: 35]
.   .   .   Child: 0  Id: a of type int [mem: Global loc: 0 size: 1] [line: 35]
.   .   Sibling: 16  Call: outnl of type void [line: 36]
.   .   Sibling: 17  Call: output of type void [line: 38]
.   .   .   Child: 0  Op: [ of type int [line: 38]
.   .   .   .   Child: 0  Id: bb of array of type int [mem: Local loc: -4 size: 101] [line: 38]
.   .   .   .   Child: 1  Op: - of type int [line: 38]
.   .   .   .   .   Child: 0  Const 66 of type int [line: 38]
.   .   .   .   .   Child: 1  Const 44 of type int [line: 38]
.   .   Sibling: 18  Call: output of type void [line: 39]
.   .   .   Child: 0  Op: * of type int [line: 39]
.   .   .   .   Child: 0  Op: [ of type int [line: 39]
.   .   .   .   .   Child: 0  Id: aa of array of type int [mem: Global loc: -2 size: 101] [line: 39]
.   .   .   .   .   Child: 1  Const 77 of type int [line: 39]
.   .   .   .   Child: 1  Op: [ of type int [line: 39]
.   .   .   .   .   Child: 0  Id: bb of array of type int [mem: Local loc: -4 size: 101] [line: 39]
.   .   .   .   .   Child: 1  Const 88 of type int [line: 39]
.   .   Sibling: 19  Call: outnl of type void [line: 40]
.   .   Sibling: 20  Call: output of type void [line: 42]
.   .   .   Child: 0  Call: dog of type int [line: 42]
.   .   .   .   Child: 0  Id: bb of array of type int [mem: Local loc: -4 size: 101] [line: 42]
.   .   Sibling: 21  Call: output of type void [line: 43]
.   .   .   Child: 0  Op: [ of type int [line: 43]
.   .   .   .   Child: 0  Id: bb of array of type int [mem: Local loc: -4 size: 101] [line: 43]
.   .   .   .   Child: 1  Const 88 of type int [line: 43]
.   .   Sibling: 22  Call: output of type void [line: 44]
.   .   .   Child: 0  Op: [ of type int [line: 44]
.   .   .   .   Child: 0  Id: bb of array of type int [mem: Local loc: -4 size: 101] [line: 44]
.   .   .   .   Child: 1  Call: dog of type int [line: 44]
.   .   .   .   .   Child: 0  Id: bb of array of type int [mem: Local loc: -4 size: 101] [line: 44]
.   .   Sibling: 23  Call: outnl of type void [line: 45]
Offset for end of global space: -102
Number of warnings: 1
Number of errors: 0
