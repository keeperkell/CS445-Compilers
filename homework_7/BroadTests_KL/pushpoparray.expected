WARNING(16): The variable 'b' seems not to be used.
Loading file: BroadTests/pushpoparray.tm
11 22 100 100
33 33
22 6776
22 89 66 89 22 90 66
Bye.
